data-otservbr-global/scripts/actions/other/exercise_training.lua

```diff
@@ -9,7 +9,7 @@ end
 local cooldown = 10
 
 function exerciseTraining.onUse(player, item, fromPosition, target, toPosition, isHotkey)
-	if not target then
+	if not target or not target:getId() then
```

Comment:
se for 'not target' não vai dar erro na segunda condição?

Reply:
> se for 'not target' não vai dar erro na segunda condição?

Tá certo, nunca vai ser "not target" pq vai entrar na primeira condição e apenas depois vai tentar dar getId, então se for not target não vai executar o not target:getId()
Mas, eu acho que depois que mudou para shared_ptr precisa checar também o not target.getId também, salvo engano é feito no script das potions

Reply:
https://github.com/opentibiabr/canary/issues/1981